lib/mnesia/test/mnesia_trans_access_test.erl

```diff
@@ -463,6 +463,111 @@ select14(Config) when is_list(Config) ->
+select_reverse14(Config) when is_list(Config) ->
+    [Node1,Node2] = Nodes = ?acquire_nodes(2, Config),
+    Tab1 = select_reverse14_ets,
+    Tab2 = select_reverse14_dets,
+    Tab3 = select_reverse14_remote,
+    Tab4 = select_reverse14_remote_dets,
+    Schemas = [[{name, Tab1}, {type, ordered_set}, {attributes, [k, v]}, {ram_copies, [Node1]}],
+	       [{name, Tab2}, {attributes, [k, v]}, {disc_only_copies, [Node1]}],
+	       [{name, Tab3}, {type, ordered_set}, {attributes, [k, v]}, {ram_copies, [Node2]}],
+	       [{name, Tab4}, {attributes, [k, v]}, {disc_only_copies, [Node2]}]],
+    [?match({atomic, ok},  mnesia:create_table(Schema)) || Schema <- Schemas],
+
+    %% Some Helpers
+    Trans = fun(Fun) -> mnesia:transaction(Fun) end,
+    Dirty = fun(Fun) -> mnesia:async_dirty(Fun) end,
+    LoopHelp = fun('$end_of_table',_) -> [];
+		  ({Recs,Cont},Fun) ->
+		       Sel = mnesia:select_reverse(Cont),
+		       Recs ++ Fun(Sel, Fun)
+	       end,
+    Loop = fun(Table,Pattern) ->
+		   Sel = mnesia:select_reverse(Table, Pattern, 1, read),
+		   Res = LoopHelp(Sel,LoopHelp),
+		   DescFun = fun(A, B) -> A > B end,
+		   case mnesia:table_info(Table, type) of
+		       ordered_set -> Res;
+		       _ -> lists:sort(DescFun, Res)
+		   end
+	   end,
+    Test =
+	fun(Tab, TabType) ->
+		OneRec = {Tab, 1, 2},
+		TwoRec = {Tab, 2, 3},
+		OnePat = [{{Tab, '$1', 2}, [], ['$_']}],
+		All = [TwoRec,OneRec],
+		AllPat = [{'_', [], ['$_']}],
+
+		?match({atomic, []}, Trans(fun() -> Loop(Tab, OnePat) end)),
+		?match({atomic, ok},   mnesia:transaction(fun() -> mnesia:write(OneRec) end)),
+		?match({atomic, ok},   mnesia:transaction(fun() -> mnesia:write(TwoRec) end)),
+		?match({atomic, [OneRec]}, Trans(fun() -> Loop(Tab, OnePat) end)),
+		?match({atomic, All}, Trans(fun() -> Loop(Tab, AllPat) end)),
+
+		{atomic,{_, ContOne}} = Trans(fun() -> mnesia:select_reverse(Tab, OnePat, 1, read) end),
+		?match({aborted, wrong_transaction}, Trans(fun() -> mnesia:select_reverse(ContOne) end)),
+		?match('$end_of_table',              Dirty(fun() -> mnesia:select_reverse(ContOne) end)),
+
+		{atomic,{_, ContAll}} = Trans(fun() -> mnesia:select_reverse(Tab, AllPat, 1, read) end),
+		?match({aborted, wrong_transaction}, Trans(fun() -> mnesia:select_reverse(ContAll) end)),
+		?match({[_], _},                     Dirty(fun() -> mnesia:select_reverse(ContAll) end)),
+
+		{atomic,_} = Trans(fun() ->
+                    case TabType of
+                        ets ->
+                            {[TwoRec], Cont} = mnesia:select_reverse(Tab, AllPat, 1, read),
+                            {[OneRec], _Cont} = mnesia:select_reverse(Cont);
+                        dets ->
+                            %% DETS will behave like a normal `select`
+                            {[OneRec], Cont} = mnesia:select_reverse(Tab, AllPat, 1, read),
+                            {[TwoRec], _Cont} = mnesia:select_reverse(Cont)
+                    end
+		end),
```

Comment:
Maybe this would be better to have tabs and spaces the same on all lines, so it would look better in the editor.